### PR TITLE
[Multiple Datasource Test]add more test for icon and aggregated view

### PR DESCRIPTION
**Pull request by yujin-emma**
### Description

add more test for icon and aggregated view
### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6724
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6723

## Testing the changes
```
  
 PASS  src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx
  DataSourceAggregatedView: read all view (displayAllCompatibleDataSources is set to true)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (37 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (13 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (12 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (16 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (11 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (10 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (9 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (8 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (8 ms)
    ✓ should render normally with local cluster configured, default datasource removed or added, and if activeDataSourceIds is present or filtered out (10 ms)
  DataSourceAggregatedView: read active view (displayAllCompatibleDataSources is set to false)
    ✓ should render normally with local cluster and active selections configured (6 ms)
    ✓ should render normally with local cluster and active selections configured (7 ms)
    ✓ should render normally with local cluster and active selections configured (6 ms)
    ✓ should render normally with local cluster and active selections configured (6 ms)
    ✓ should render normally with local cluster and active selections configured (5 ms)
    ✓ should render normally with local cluster and active selections configured (6 ms)
    ✓ should render normally with local cluster and active selections configured (5 ms)
    ✓ should render normally with local cluster and active selections configured (5 ms)
  DataSourceAggregatedView empty state test with local cluster hiding
    ✓ should render warning when no data sources added (4 ms)
    ✓ should render warning when no data sources added (2 ms)
    ✓ should render warning when no data sources added (4 ms)
    ✓ should render warning when no data sources added (2 ms)
  DataSourceAggregatedView empty state test due to filter out with local cluster hiding
    ✓ should render warning when no data sources added (2 ms)
    ✓ should render warning when no data sources added (2 ms)
  DataSourceAggregatedView error state test no matter hide local cluster or not
    ✓ should render error state when catch error (2 ms)
    ✓ should render error state when catch error (2 ms)
    ✓ should render error state when catch error (1 ms)
    ✓ should render error state when catch error (1 ms)


 PASS  src/plugins/data_source_management/public/components/custom_database_icon/empty_icon.test.tsx
  Test on empty icon
    ✓ should render the component normally (7 ms)

 PASS  src/plugins/data_source_management/public/components/custom_database_icon/error_icon.test.tsx
  Test on empty icon
    ✓ should render the component normally (7 ms)

```

## Changelog
- fix: Add more test for icon and aggregated view

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Multiple Datasource] Add empty state component for no connected data source ([#6499](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6499))
 - [Multiple Datasource] Add popover for empty state and redirect to data source management page([#6514](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6514))
 - [Multiple Datasource] Update empty state font size and footer button size to small ([6549](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6549))
+- [Multiple Datasource] Add UT for empty, error icon and aggregated view ([#6729](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6729))
```
Comment (BionIT) [inline CHANGELOG.md:92]: I think we stopped using this md for change log

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx [hunk 2]
```diff
@@ -269,3 +270,151 @@ describe('DataSourceAggregatedView: read active view (displayAllCompatibleDataSo
...
+          displayAllCompatibleDataSources={displayAllCompatibleDataSources}
+          uiSettings={uiSettings}
+          activeDataSourceIds={activeDataSourceIds}
+          dataSourceFilter={filter}
+        />
+      );
+
+      expect(component).toMatchSnapshot();
+      await nextTick();
+      expect(toasts.add).toBeCalled();
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:342]: can we verify the toast message?

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx [hunk 3]
```diff
@@ -269,3 +270,151 @@ describe('DataSourceAggregatedView: read active view (displayAllCompatibleDataSo
...
+      expect(component.state('showEmptyState')).toBe(true);
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:343]: can we verify the empty component is mounted?

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx [hunk 4]
```diff
@@ -269,3 +270,151 @@ describe('DataSourceAggregatedView: read active view (displayAllCompatibleDataSo
...
+  let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
+  let client: SavedObjectsClientContract;
+  const { toasts } = notificationServiceMock.createStartContract();
+  const uiSettings = uiSettingsServiceMock.createStartContract();
+  const application = applicationServiceMock.createStartContract();
+  const nextTick = () => new Promise((res) => process.nextTick(res));
+
+  beforeEach(() => {
+    client = {
+      find: jest.fn().mockResolvedValue([]),
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:284]: looks like we return empty array no matter what, and are we testing the filter at all?

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx [hunk 5]
```diff
@@ -269,3 +270,151 @@ describe('DataSourceAggregatedView: read active view (displayAllCompatibleDataSo
...
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:358]: how did we simulate the error state?

File: CHANGELOG.md (hunk 1 above)
Comment (yujin-emma) [reply CHANGELOG.md:92]: Oh,  I do not know that, we do not use the change log anymore?

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx (hunk 2 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:342]: sure, updated with
```
expect(toasts.add).toHaveBeenCalledTimes(1);
      expect(toasts.add.mock.calls[0][0]).toEqual({
        "color": "warning",
        "text": expect.any(Function),
        "title": "No data sources connected yet. Connect your data sources to get started."
      });
      expect(component.state('showEmptyState')).toBe(true);
```

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx (hunk 3 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:343]: sure, updated with:
```
      expect(component.find("NoDataSource").exists()).toBe(true);
```

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx (hunk 5 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:358]: use this mock function can mock the error state:
```
    mockErrorResponseForSavedObjectsCalls(client, 'find');

```

Comment (BionIT) [reply src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.test.tsx:438]: in this case, do we still need to initialize client object here?

**Review by BionIT — APPROVED** [review]

**Review by ZilongX — APPROVED** [review]

**Comment by ZilongX** [comment]
Looks like we're missing a DCO sign off

**Pull request merged**